src/prefect/cli/orion.py

```diff
@@ -150,6 +152,42 @@ async def start(
     app.console.print("Orion stopped!")
 
 
+@orion_app.command()
+async def status(
+    wait: bool = typer.Option(
+        False,
+        "--wait",
+        help="Runs health check loop until successful or optional timeout value is exceeded.",
+    ),
+    timeout: int = typer.Option(
+        None, "--timeout", help="Optional Timeout value in seconds"
+    ),
+):
+    """Verify Connection status with Orion API"""
+    async with get_client() as client:
+        status_check = True
+        start_time = pendulum.now()
+        while status_check == True:
+            response = await client.api_healthcheck()
+            if not response:
+                print("Orion Healthy")
+                status_check = False
+            else:
+                if wait == False:
+                    print(response)
+                    status_check = False
+                else:
+                    if not timeout:
+                        print(f"{response} retrying")
+                    else:
+                        timeout_diff = pendulum.now() - start_time
+                        if timeout_diff.seconds > timeout:
+                            print(f"Timeout exceeded, {response}")
+                            status_check = False
+                        else:
+                            print(f"{response} retrying")
```

Comment:
I'd use `anyio.move_on_after` to enforce your timeout instead, you can also just break on a healthy response e.g.

```python

import anyio
import random
import contextlib


async def main():
    timeout = None
    with (
        anyio.move_on_after(timeout) if timeout else contextlib.nullcontext()
    ) as timeout_scope:
        while True:
            if not random.choice(range(10)):
                break
            await anyio.sleep(1)

    if timeout and timeout_scope.cancel_called:
        print("Timed out!")
        return -1

    print("Healthy!")
```

Reply:
I updated the code to use move_on_after and it works pretty well, it is generating a runtime error "RuntimeError: The connection pool was closed while 1 HTTP requests/responses were still in-flight." that I'm not sure if we want to suppress since this would likely be expected when it times out while processing a request